src/components/ApplicationGroup/AppGroupDetailsRoute.tsx

```diff
@@ -215,6 +215,9 @@ export default function AppGroupDetailsRoute({ isSuperAdmin }: AppGroupAdminType
                     .sort(sortOptionsByLabel),
             )
         }
+        if (selectedGroupFilter.length) {
+            setSelectedAppList(appListOptions.filter((app) => selectedGroupFilter[0].appIds.indexOf(+app.value) >= 0))
```

Comment:
- Fix the state issue as discussed
- 
```suggestion
            setSelectedAppList(appListOptions.filter((app) => selectedGroupFilter[0].appIds.includes(+app.value)))
```